crates/solver/src/interactions/allowances.rs

```diff
@@ -234,6 +234,26 @@ fn is_batch_error(err: &ExecutionError) -> bool {
     }
 }
 
+/// An allowance manager that always reports no allowances.
+pub struct NoAllowanceManaging;
```

Comment:
Makes sense. I had originally put it here because I thought that `async-trait` wasn't a dependency of `driver`, and thought it was "cleaner" to avoid pulling in an additional dependency for a "mock" boundary implementation.

However, it looks like `async-trait` is already a dependency, so my argument is moot.